packages/wallet/contract/test/test-psm.js

```diff
@@ -0,0 +1,226 @@
+// @ts-check
+import { test as anyTest } from '@agoric/zoe/tools/prepare-test-env-ava.js';
+
+import { AmountMath } from '@agoric/ertp';
+import { buildRootObject as buildPSMRootObject } from '@agoric/vats/src/core/boot-psm.js';
+import '@agoric/vats/src/core/types.js';
+import {
+  mockDProxy,
+  mockPsmBootstrapArgs,
+} from '@agoric/vats/tools/boot-test-utils.js';
+import '@agoric/zoe/exported.js';
+import { E } from '@endo/eventual-send';
+import { makeDefaultTestContext } from './contexts.js';
+
+/**
+ * @type {import('ava').TestFn<Awaited<ReturnType<makeDefaultTestContext>>
+ * & {consume: import('@agoric/inter-protocol/src/proposals/econ-behaviors.js').EconomyBootstrapPowers['consume']}>
+ * }
+ */
+const test = anyTest;
+
+const address = 'psmTestAddress';
+
+const makePsmTestSpace = async log => {
+  const psmParams = {
+    anchorAssets: [{ denom: 'ibc/usdc1234', keyword: 'AUSD' }],
+    economicCommitteeAddresses: [address],
+    argv: { bootMsg: {} },
+  };
+
+  const psmVatRoot = await buildPSMRootObject(
+    {
+      logger: log,
+      D: mockDProxy,
+    },
+    psmParams,
+  );
+  psmVatRoot.bootstrap(...mockPsmBootstrapArgs(log));
+
+  // @ts-expect-error cast
+  return /** @type {ChainBootstrapSpace} */ (psmVatRoot.getPromiseSpace());
+};
+
+test.before(async t => {
+  // @ts-expect-error cast
+  t.context = await makeDefaultTestContext(t, makePsmTestSpace);
+});
+
+/** @type {<T>(subscriber: Subscriber<T>) => Promise<T>} */
+const currentState = async subscriber => {
+  const record = await subscriber.subscribeAfter();
+  return record.head.value;
+};
+/**
+ * @param {import('../src/smartWallet.js').PursesRecord} pursesRecord
+ * @param {Brand} brand
+ */
+const purseBalance = (pursesRecord, brand) => {
+  const purseDesc = pursesRecord.find(desc => desc.brand === brand);
+  assert(purseDesc, `${brand} not found in record`);
+  return purseDesc.purseBalance.value;
+};
+
+test('null swap', async t => {
+  const { anchor } = t.context;
+  const { agoricNames, board } = await E.get(t.context.consume);
+  const minted = await E(agoricNames).lookup('brand', 'IST');
+
+  const wallet = await t.context.simpleProvideWallet(address);
+  const pursesSubscriber = wallet.getDataFeeds().purses.subscriber;
+
+  const psmInstance = await E(agoricNames).lookup('instance', 'psm');
+
+  /** @type {import('../src/invitations.js').ContractInvitationSpec} */
+  const invitationSpec = {
+    source: 'contract',
+    instance: psmInstance,
+    publicInvitationMaker: 'makeGiveStableInvitation',
+  };
+  const publishingMarshal = await E(board).getPublishingMarshaller();
+  const offerSpecCapData = publishingMarshal.serialize(
+    harden({
+      invitationSpec,
+      proposal: {
+        // empty amounts
+        give: { In: AmountMath.makeEmpty(minted) },
+        want: { Out: anchor.makeEmpty() },
+      },
+    }),
+  );
+
+  // @ts-expect-error faulty generic typedef
+  wallet.executeOffer(offerSpecCapData);
```

Comment:
`await`? ah. no. We shouldn't rely on getting a promise for the result. Seems worth documenting with `void` if not a comment.

Reply:
you mean document at the callsite? the docs on `executeOffer` already say it's a synchronous function:

>`      * @returns {void} when the offer has been sent to Zoe; payouts go into this wallet's purses`